redbot/core/modlog.py

```diff
@@ -174,6 +174,14 @@
         await _conf.custom(_CASETYPES).set(all_casetypes)
         await _conf.schema_version.set(3)
 
+    if from_version < 4 <= to_version:
+        # set latest_case_number
+        for guild_id, cases in (await _conf.custom(_CASES).all()).items():
+            if cases:
+                await _conf.guild(
+                    cast(discord.Guild, discord.Object(id=guild_id))
+                ).latest_case_number.set(max(map(int, cases.keys())))
+
```

Comment:
you need to set the schema version 4 in this block

Reply:
Good point, forgot about that :+1: